app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -112,6 +112,11 @@ def index_relation_environment(query)
     param_group :repo
     def create
       repo_params = repository_params
+      unless RepositoryTypeManager.creatable_by_user?(repo_params[:content_type])
+        msg = _("Invalid params provided - content_type must be one of %s") % RepositoryTypeManager.creatable_repository_types.join(",")
```

Comment:
`createable_repository_types` returns a hash which doesn't have a join method and blows up if you pass in `file` as the type